src/Entity/LearningMaterial.php

```diff
@@ -677,9 +677,9 @@ public function getSessions()
      */
     public function getValidationGroups()
     {
-        if ('' !== trim($this->getCitation())) {
+        if ($this->getCitation() && strlen(trim($this->getCitation())) > 0) {
```

Comment:
super edge case, but falsy values such as `0` will fall through the cracks here. i think we need to be more diligent here, in order to prevent the citation/link LMs to be incorrectly validated as files.

should be:
`if (null !== $this->getCitation() && strlen(trim($this->getCitation())) > 0) {`

Reply:
It would be difficult to store the integer 0 in the DB, but I agree. Fixed.

Reply:
more concerned about string `"0"`, which is also falsy. b/c PHP.